manifests/init.pp

```diff
@@ -162,6 +165,16 @@
     $iftemplate = template('network/ifcfg-eth.erb')
   }
 
+  if $flush {
+    exec { 'network-flush':
+      user        => 'root',
+      command     => "/usr/sbin/ip addr flush dev ${interface}",
```

Comment:
If I don't have any path, Puppet will fail with error `'ip addr flush dev eth0' is not qualified and no path was specified. Please qualify the command or specify a path.'`.

It's been around for a while, definitely with Puppet 4 or Puppet 3.7. Unfortunately this doesn't show up in spec tests, I think because it's implemented in provider rather than type itself.